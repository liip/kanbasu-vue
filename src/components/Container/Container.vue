<template>
  <div :class="classNames">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'Container',

  props: {
    variant: {
      type: String,
    },
  },

  computed: {
    classNames() {
      return ['container', { [`container--${this.variant}`]: !!this.variant }];
    },
  },
};
</script>
