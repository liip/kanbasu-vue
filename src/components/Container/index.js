import Container from './Container.vue';

Container.install = function(Vue) {
  Vue.component(Container.name, Container);
};

export default Container;
