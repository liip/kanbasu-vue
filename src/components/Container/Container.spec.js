import { shallowMount } from '@vue/test-utils';
import Container from './Container.vue';

describe('Container.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Container);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with variant', () => {
    const wrapper = shallowMount(Container, {
      propsData: {
        variant: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
