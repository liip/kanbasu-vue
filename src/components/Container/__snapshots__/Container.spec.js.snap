// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container.vue renders 1`] = `<div class="container"></div>`;

exports[`Container.vue renders with variant 1`] = `<div class="container container--small"></div>`;
